src/components/common/Dialogs/PermissionManagementDialog/PermissionManagementForm.tsx

```diff
@@ -0,0 +1,371 @@
+import React, { useCallback, useMemo } from 'react';
+import { defineMessages } from 'react-intl'; // FormattedMessage
+import {
+  ColonyRole,
+  Id,
+  // VotingReputationVersion,
+} from '@colony/colony-js';
+import { FormState } from 'react-hook-form';
+
+import { InputLabel, Select, Annotations } from '~shared/Fields'; // ForceToggle
+import { DialogSection } from '~shared/Dialog';
+import { Heading3 } from '~shared/Heading';
+import PermissionRequiredInfo from '~shared/PermissionRequiredInfo';
+import SingleUserPicker, {
+  filterUserSelection,
+} from '~shared/SingleUserPicker';
+import { ItemDataType } from '~shared/OmniPicker';
+import UserAvatar from '~shared/UserAvatar';
+
+// import MotionDomainSelect from '~dashboard/MotionDomainSelect';
+
+import { Address, Colony, User } from '~types/index';
+import { notNull } from '~utils/arrays';
+// import { useTransformer, useEnabledExtensions} from '~hooks';
+// import { getAllUserRolesForDomain } from '~redux/transformers';
+import { sortBy } from '~utils/lodash';
+
+import { availableRoles } from './constants';
+import { FormValues } from './PermissionManagementDialog';
+import PermissionManagementCheckbox from './PermissionManagementCheckbox';
+
+import styles from './PermissionManagementDialog.css';
+
+const displayName = `common.ColonyHome.PermissionManagementDialog.PermissionManagementForm`;
+
+const MSG = defineMessages({
+  title: {
+    id: `${displayName}.title`,
+    defaultMessage: 'Permissions',
+  },
+  domain: {
+    id: `${displayName}.domain`,
+    defaultMessage: 'Team',
+  },
+  permissionsLabel: {
+    id: `${displayName}.permissionsLabel`,
+    defaultMessage: 'Permissions',
+  },
+  annotation: {
+    id: `${displayName}.annotation`,
+    defaultMessage: 'Explain why you’re making these changes (optional)',
+  },
+  selectUser: {
+    id: `${displayName}.selectUser`,
+    defaultMessage: 'Member',
+  },
+  userPickerPlaceholder: {
+    id: `${displayName}.userPickerPlaceholder`,
+    defaultMessage: 'Search for a user or paste wallet address',
+  },
+  cannotCreateMotion: {
+    id: `${displayName}.cannotCreateMotion`,
+    defaultMessage: `Cannot create motions using the Governance v{version} Extension. Please upgrade to a newer version (when available)`,
+  },
+});
+
+interface Props {
+  colony: Colony;
+  currentUserRoles: ColonyRole[];
+  domainId: number;
+  rootAccounts: Address[];
+  userDirectRoles: ColonyRole[];
+  currentUserRolesInRoot: ColonyRole[];
+  userInheritedRoles: ColonyRole[];
+  onDomainSelected: (domain: number) => void;
+  onChangeSelectedUser: React.Dispatch<React.SetStateAction<string>>;
+  onMotionDomainChange: (domain: number) => void;
+  inputDisabled: boolean;
+  userHasPermission: boolean;
+  values: FormValues;
+}
+
+const supRenderAvatar = (address: string, item: ItemDataType<User>) => (
+  <UserAvatar address={address} user={item} size="xs" notSet={false} />
+);
+
+const PermissionManagementForm = ({
```

Comment:
This is going to be painful to write in once we get to it.

Most likely it'll also need a bit of refactoring